spec/controllers/coronavirus_form/check_answers_controller_spec.rb

```diff
@@ -28,13 +29,45 @@
 
     it "saves the form response to the database" do
       session["attribute"] = "key"
+      session["medical_equipment"] = I18n.t("coronavirus_form.questions.medical_equipment.options.option_yes.label")
+      session["contact_details"] = { "contact_name" => "Harry Potter" }
       post :submit
 
       expect(FormResponse.first.form_response).to eq(
-        [%w[attribute key], %w[reference_number abc]],
+        "attribute" => "key",
+        "contact_details" => { "contact_name" => "Harry Potter" },
+        "medical_equipment" => "Yes",
+        "reference_number" => "abc",
       )
     end
 
+    context "when unwanted fields are present in the session" do
+      it "excludes those fields from the FormResponse" do
+        data = valid_data
+        data.map { |k, v| session[k.to_s] = v }
```

Comment:
What is this line doing?

Reply:
Oh! Are you setting the session values?

Reply:
This is adding the valid_data from the schema helper to the session, with stringified keys.

Reply:
Why does it need `to_s`?  Does `session.merge!(valid_data)` work?

Reply:
Yeah that works. That's way better, thanks!